### PR TITLE
Repo cleanup

### PR DESCRIPTION
**Pull request by farisashai**
# Description
Cleanup and some improvements

I wouldn't merge this until after all of the major PRs are closed so the fixes that apply there too can be added in without introducing more merge conflicts. 

## Changes

- Removed unnecessary scaffolding code from when I initialized this repo for navbar props (we can directly intercept the pageProps to see if the user is logged in and don't have to store any state)
- Clear cookie function to support future flexibility if we add additional cookie types
- Corrected some data types and made components more concise
- Renamed PermissionService.allUserTypes to be more descriptive in page access restrictions
- Blocked user.state cannot log in or restricted user.accessType
- 404 page is not relevant enough to need user data, just redirect to home page and let them recover from there
- Store dropdown divider in a constant instead of being a magic variable
- Added support for the project as a Progressive Web App. Biggest change is that a user can click add to home screen on mobile and it will open as a native app instead of in a web browser, but PWAs also have lots of other benefits for improving caching and native experiences.

# Type of Change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Logistics Change (A change to a README, description, or dev workflow setup like
      linting/formatting)
- [ ] Continuous Integration Change (Related to deployment steps or continuous integration
      workflows)
- [ ] Other: (Fill In) 

# Testing

I have tested that my changes fully resolve the linked issue ...

- [x] locally on Desktop.
- [ ] locally on mobile - use https://ngrok.io to get a copy on a mobile device
- [x] on the live deployment preview on Desktop.
- [ ] on the live deployment preview on Mobile.
- [ ] I have run and passed all new and existing Cypress tests. Add screenshots below.

# Checklist

- [x] I have performed a self-review of my own code.
- [x] I have followed the style guidelines of this project.
- [x] I have documented my code's `src/lib` functions and commented hard to understand areas
      anywhere else.
- [x] My changes produce no new warnings.

### CODE REVIEW

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -6,37 +6,14 @@ import 'react-toastify/dist/ReactToastify.css';
...
-  useEffect(() => {
-    if (pageProps?.user) {
-      setUser(pageProps?.user);
-      return;
-    }
-    const userCookie = CookieService.getClientCookie(CookieType.USER);
-    if (userCookie) {
-      setUser(JSON.parse(userCookie));
-    }
-  }, [pageProps?.user]);
```
Comment (SheepTester) [inline src/pages/_app.tsx:38]: regression: removing this line undoes a change (which came from the storefront PR) that adds the signed-in navbar to the 404 page

main branch (for https://v2.members.acmucsd.com/store/collection/aca1d17d-7002-46fb-808c-f080703edd4c):

image


this branch:

image

Comment (farisashai) [reply src/pages/_app.tsx:38]: we can remove that feature i think

File: src/pages/index.tsx [hunk 2]
```diff
@@ -51,8 +51,10 @@ export default PortalHomePage;
 
 const getServerSidePropsFunc: GetServerSideProps = async ({ req, res }) => {
   const authToken = CookieService.getServerCookie(CookieType.ACCESS_TOKEN, { req, res });
-  const events = await EventAPI.getAllEvents();
-  const attendances = await EventAPI.getAttendancesForUser(authToken);
+  const getEvents = EventAPI.getAllEvents();
+  const getAttendances = EventAPI.getAttendancesForUser(authToken);
+
+  const [events, attendances] = await Promise.all([getEvents, getAttendances]);
```
Comment (alexzhang1618) [inline src/pages/index.tsx]: Don't think we need these changes anymore since I've added batching into the index page

**Review by alexzhang1618 — APPROVED** [review]
LGTM for whenever we want this merged

**Pull request merged**